### ISSUE TITLE
[no-use-before-define] Add option to ignore interfaces

### ISSUE THREAD

**Issue by avahe-kellenberger** [typescript-eslint/typescript-eslint#1262]
**Repro**

```json
{
  "rules": {
    "@typescript-eslint/no-use-before-define": [
        "error",
        {
            classes: true,
            functions: false,
            typedefs: false,
        },
    ]
  }
}
```

```TS
class A implements B {}

interface B {
  a: string
}
```

**Expected Result**
No error, since interfaces are hoisted.

**Actual Result**
'B' was used before it was defined. eslint(@typescript-eslint/no-use-before-define)

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.6.1` |
| `@typescript-eslint/parser`        | `2.6.1` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `13.2.0` |
| `npm`                              | `6.12.1` |

**Comment by bradzacher** [comment]
It's just generally recommended to just not use this rule.
It's got really bad support for TS features across the board, and it doesn't do anything useful that the TS typechecker doesn't already do.

`typedefs` should cover `interface` and `type` declarations, but it doesn't actually do anything (#249), and nobody has bothered to fix it.

The bigger problem here is that the scope analysis is not in a great state, so the rule will never work completely correctly..

that all being said - I won't block anyone that wants to add an option to the rule..

**Comment by bradzacher** [comment]
Merging into #1856
